src/utils/agent.ts

```diff
@@ -102,7 +102,12 @@ const Projects = {
 }
 
 const Opportunities = {
-    list: () => requests.get<Opportunity[]>('opportunities'),
+    list: (params?: URLSearchParams) => requests.get<Opportunity[]>('opportunities', params),
+    post: (body: Opportunity) => requests.post<Opportunity>('opportunities', body),
+    getCount: () => requests.get<Object>('opportunities/count'),
+    getDetails: (id: string) => requests.get<Opportunity>(`opportunities/${id}`),
+    update: (id: string, body: Opportunity) => requests.put<Opportunity>(`opportunities/${id}`, body),
```

Comment:
id should be number

Reply:
body is suppose to have all the interface properties set to optional because when send a put request to an api you only want to change specific fields and not everything.

check this out, maybe you'll learn something great with these utility types
https://www.typescriptlang.org/docs/handbook/utility-types.html#partialtype